client/signup/steps/theme-selection/index.jsx

```diff
@@ -33,6 +33,12 @@ module.exports = React.createClass( {
 		};
 	},
 
+	componentWillReceiveProps( nextProps ) {
+		if ( nextProps.signupDependencies !== this.props.signupDependencies && nextProps.signupDependencies.theme === this.props.signupDependencies.theme ) {
```

Comment:
Actually, that may make the condition never true (?). I was going to write this version in the first place. How about:

```
const { themes, surveyQuestion, designType } = nextProps.signupDependencies;
if ( themes !== this.props.signupDependencies.themes || surveyQuestion !== this.props.signupDependencies.surveyQuestion || designType !== this.props.signupDependencies.designType ) {
```